src/App.tsx

```diff
@@ -165,38 +201,120 @@ class Board extends React.Component<BoardProps, BoardState> {
         }
     }
 
+    public handleGlobalKeyUp = (e: KeyboardEvent) => {
+        if (e.keyCode === 91) {
+            e.preventDefault();
+            const newState = _.cloneDeep(this.state);
+            newState.continueHighlighting = false;
+            this.setState(newState);
+        }
+    }
+
+    public handleGlobalMouseDown = (e: MouseEvent) => {
+        if (!e.defaultPrevented && !this.state.continueHighlighting) {
+            e.preventDefault();
+            this.deselectSquares();
+        }
+    }
+
+    public handleGlobalMouseUp = (e: MouseEvent) => {
+        if (!e.defaultPrevented) {
+            e.preventDefault();
+            const newState = _.cloneDeep(this.state);
+            newState.highlighting = false;
+            this.setState(newState);
+        }
+    }
+
+    public handleSquareMouseDown = (i: SquareAddress) => (e: MouseEvent) => {
+        e.preventDefault();
+        const newState = _.cloneDeep(this.state);
+        newState.highlighting = true;
+        if (this.state.continueHighlighting) {
+            newState.highlights.add(i);
+        } else {
+            newState.highlights = new Set([i]);
+        }
+        this.setState(newState);
+    }
+
+    public handleSquareMouseOver = (i: SquareAddress) => (e: MouseEvent) => {
+        e.preventDefault();
+        if (this.state.highlighting) {
+            const newState = _.cloneDeep(this.state);
+            newState.highlights.add(i);
+            this.setState(newState);
+        }
+    }
+
+    public handleSquareMouseUp = (i: SquareAddress) => (e: MouseEvent) => {
+        e.preventDefault();
+        const newState = _.cloneDeep(this.state);
+        newState.highlighting = false;
+        this.setState(newState);
```

Comment:
Nice. This is beautiful.

Think about why your intuition was guiding you to a "select div" library. Compare your original intuition with the simplicity of this solution. Update your intuition around what you can build manually for the future.